Mobile Buy SDK/Mobile Buy SDK/Operations/BUYCheckoutOperation.m

```diff
@@ -144,25 +110,11 @@ - (NSOperation *)createBeginOperation
 	}];
 }
 
-- (NSOperation *)createPollOperation
-{
-	BUYRequestOperation *operation = (BUYRequestOperation *)[self.client getCompletionStatusOfCheckoutWithToken:self.checkoutToken start:NO completion:^(BUYStatus status, NSError *error) {
-		if (status != BUYStatusComplete) {
-			[self finishWithError:error];
-		}
-	}];
-	operation.pollingHandler = ^BOOL (NSDictionary *json, NSHTTPURLResponse *response, NSError *error) {
-		return response.statusCode == BUYStatusProcessing;
-	};
-	
-	return operation;
-}
-
-- (NSOperation *)createGetOperation
+- (NSOperation *)createStatusOperation
 {
-	return [self.client getCheckoutWithToken:self.checkoutToken start:NO completion:^(BUYCheckout *checkout, NSError *error) {
+	return [self.client pollCompletionStatusAndGetCheckoutWithToken:self.checkoutToken start:NO completion:^(BUYCheckout *checkout, NSError *error) {
 		if (checkout) {
```

Comment:
I personally think its easier to read when we check for `cancelled` state in `finishWith...` methods. Its also less error prone since there's no chance that someone can forget to check for `cancelled` before calling `finishWith...`.